src/devices/machine/smc91c9x.cpp

```diff
@@ -452,21 +731,19 @@ READ16_MEMBER( smc91c9x_device::read )
 
 WRITE16_MEMBER( smc91c9x_device::write )
 {
-	//  uint16_t olddata;
-
 	/* determine the effective register */
 	offset %= 8;
 	if (offset != EREG_BANK)
 		offset += 8 * (m_reg[EREG_BANK] & 7);
 
 	/* update the data generically */
-	//  olddata = m_reg[offset];
+
+	if (LOG_ETHERNET && offset != 7 && offset < sizeof(m_reg))
+		logerror("%s:smc91c9x_w(%s) = [%04X]<-%04X & (%04X & %04X)\n", (machine().describe_context()).c_str(), ethernet_regname[offset], offset, data, mem_mask , m_regmask[offset]);
```

Comment:
Once again, please don't use `.c_str()` with `logerror` - it's not unnecessary clutter and shouldn't be encouraged.

Reply:
Done in the new PR.